src/core/handlers/keep-if.ts

```diff
@@ -0,0 +1,71 @@
+/*
+* keep-if.ts
+*
+* Copyright (C) 2022 by RStudio, PBC
+*
+*/
+
+import { LanguageCellHandlerContext, LanguageHandler } from "./types.ts";
+import { baseHandler, install } from "./base.ts";
+import {
+  isDocxOutput,
+  isHtmlCompatible,
+  isJavascriptCompatible,
+  isMarkdownOutput,
+  isPdfOutput,
+} from "../../config/format.ts";
+import { QuartoMdCell } from "../lib/break-quarto-md.ts";
+import { asMappedString, mappedConcat } from "../lib/mapped-text.ts";
+
+const keepIfHandler: LanguageHandler = {
+  ...baseHandler,
+
+  type: "directive",
+  stage: "post-engine",
+
+  languageName: "keep-if",
+
+  defaultOptions: {},
+
+  cell(
+    handlerContext: LanguageCellHandlerContext,
+    cell: QuartoMdCell,
+  ) {
+    // FIXME there's probably a more elegant way to do this...
```

Comment:
Let's discuss approach to this. We definitely need to aggregate as is being done here and we _may_ want to support some different macros as we do for. yaml validation.